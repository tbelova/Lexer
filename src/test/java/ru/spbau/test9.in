sum2 (a, b, mod) {
	return (a + b) % m;
}

sum3(a, b, c, m) {
	return sum2(sum2(a, b, m), m);
}

read x;
read y;
read z;
read mod;

write sum3(x, y, z, mod); 

