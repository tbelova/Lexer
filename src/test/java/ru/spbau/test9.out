PROGRAM((1,0)(16,4))
   STATEMENTS((1,0)(14,24))
      FUNCTION((1,0)(3,0))
         IDENT((1,0)(1,3))[sum2]
         IDENTS((1,6)(1,14))
            IDENT((1,6)(1,6))[a]
            IDENTS((1,9)(1,14))
               IDENT((1,9)(1,9))[b]
               IDENTS((1,12)(1,14))
                  IDENT((1,12)(1,14))[mod]
         BLOCK((1,17)(3,0))
            STATEMENTS((2,1)(2,19))
               STATEMENT((2,1)(2,19))
                  RETURN((2,1)(2,18))
                     EXPRESSION((2,8)(2,18))
                        REM((2,8)(2,18))
                           SUM((2,9)(2,13))
                              IDENT((2,9)(2,9))[a]
                              IDENT((2,13)(2,13))[b]
                           IDENT((2,18)(2,18))[m]
      STATEMENTS((5,0)(14,24))
         FUNCTION((5,0)(7,0))
            IDENT((5,0)(5,3))[sum3]
            IDENTS((5,5)(5,14))
               IDENT((5,5)(5,5))[a]
               IDENTS((5,8)(5,14))
                  IDENT((5,8)(5,8))[b]
                  IDENTS((5,11)(5,14))
                     IDENT((5,11)(5,11))[c]
                     IDENTS((5,14)(5,14))
                        IDENT((5,14)(5,14))[m]
            BLOCK((5,17)(7,0))
               STATEMENTS((6,1)(6,30))
                  STATEMENT((6,1)(6,30))
                     RETURN((6,1)(6,29))
                        EXPRESSION((6,8)(6,29))
                           FUNCTION_CALL((6,8)(6,29))
                              IDENT((6,8)(6,11))[sum2]
                              EXPRESSIONS((6,13)(6,28))
                                 EXPRESSION((6,13)(6,25))
                                    FUNCTION_CALL((6,13)(6,25))
                                       IDENT((6,13)(6,16))[sum2]
                                       EXPRESSIONS((6,18)(6,24))
                                          EXPRESSION((6,18)(6,18))
                                             IDENT((6,18)(6,18))[a]
                                          EXPRESSIONS((6,21)(6,24))
                                             EXPRESSION((6,21)(6,21))
                                                IDENT((6,21)(6,21))[b]
                                             EXPRESSIONS((6,24)(6,24))
                                                EXPRESSION((6,24)(6,24))
                                                   IDENT((6,24)(6,24))[m]
                                 EXPRESSIONS((6,28)(6,28))
                                    EXPRESSION((6,28)(6,28))
                                       IDENT((6,28)(6,28))[m]
         STATEMENTS((9,0)(14,24))
            STATEMENT((9,0)(9,6))
               READ_STATEMENT((9,0)(9,5))
                  IDENT((9,5)(9,5))[x]
            STATEMENTS((10,0)(14,24))
               STATEMENT((10,0)(10,6))
                  READ_STATEMENT((10,0)(10,5))
                     IDENT((10,5)(10,5))[y]
               STATEMENTS((11,0)(14,24))
                  STATEMENT((11,0)(11,6))
                     READ_STATEMENT((11,0)(11,5))
                        IDENT((11,5)(11,5))[z]
                  STATEMENTS((12,0)(14,24))
                     STATEMENT((12,0)(12,8))
                        READ_STATEMENT((12,0)(12,7))
                           IDENT((12,5)(12,7))[mod]
                     STATEMENTS((14,0)(14,24))
                        STATEMENT((14,0)(14,24))
                           WRITE_STATEMENT((14,0)(14,23))
                              EXPRESSION((14,6)(14,23))
                                 FUNCTION_CALL((14,6)(14,23))
                                    IDENT((14,6)(14,9))[sum3]
                                    EXPRESSIONS((14,11)(14,22))
                                       EXPRESSION((14,11)(14,11))
                                          IDENT((14,11)(14,11))[x]
                                       EXPRESSIONS((14,14)(14,22))
                                          EXPRESSION((14,14)(14,14))
                                             IDENT((14,14)(14,14))[y]
                                          EXPRESSIONS((14,17)(14,22))
                                             EXPRESSION((14,17)(14,17))
                                                IDENT((14,17)(14,17))[z]
                                             EXPRESSIONS((14,20)(14,22))
                                                EXPRESSION((14,20)(14,22))
                                                   IDENT((14,20)(14,22))[mod]
